lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -1267,62 +1237,54 @@ ValueDecl *DerivedConformance::deriveEncodable(TypeChecker &tc,
   // diagnostics, then potentially collect notes. If we succeed in
   // synthesizing Encodable, we can cancel the transaction and get rid of the
   // fake failures.
-  auto diagnosticTransaction = DiagnosticTransaction(tc.Context.Diags);
-  tc.diagnose(target, diag::type_does_not_conform, target->getDeclaredType(),
-              encodableType);
-  tc.diagnose(requirement, diag::no_witnesses, diag::RequirementKind::Func,
-              requirement->getFullName(), encodableType, /*AddFixIt=*/false);
+  auto diagnosticTransaction = DiagnosticTransaction(TC.Context.Diags);
+  TC.diagnose(Nominal, diag::type_does_not_conform, Nominal->getDeclaredType(),
```

Comment:
Don't we want this diagnostic to point at the place where the conformance was declared, rather than always pointing at the nominal type declaration?